From 3fa71fac485e58344119808cf9f5ee8255bfdf12 Mon Sep 17 00:00:00 2001
From: Diego Rondini <diego.rondini@kynetics.com>
Date: Thu, 1 Jun 2017 15:34:01 +0200
Subject: [PATCH 3/4] server_hawkbit: add artifactsstorage parameter

Add artifactsstorage parameter to allow to save artifact files to a
specified directory instead of installing them directly.
---
 suricatta/server_hawkbit.c | 12 ++++++++++--
 suricatta/server_hawkbit.h |  1 +
 2 files changed, 11 insertions(+), 2 deletions(-)

diff --git a/suricatta/server_hawkbit.c b/suricatta/server_hawkbit.c
index d70e671..6c967b4 100644
--- a/suricatta/server_hawkbit.c
+++ b/suricatta/server_hawkbit.c
@@ -77,6 +77,7 @@ static struct option long_options[] = {
     {"retry", required_argument, NULL, 'r'},
     {"retrywait", required_argument, NULL, 'w'},
     {"proxy", optional_argument, NULL, 'y'},
+    {"artifactsstorage", optional_argument, NULL, 's'},
     {NULL, 0, NULL, 0}};
 
 static unsigned short mandatory_argument_count = 0;
@@ -1480,7 +1481,8 @@ void suricatta_print_help(void)
 	    "\t  -w, --retrywait     Time to wait prior to retry and "
 	    "resume a download (default: %ds).\n"
 	    "\t  -y, --proxy         Use proxy. Either give proxy URL, else "
-	    "{http,all}_proxy env is tried.\n",
+	    "{http,all}_proxy env is tried.\n"
+	    "\t  -s, --artifactsstorage Specify artifacts storage directory.\n",
 	    DEFAULT_POLLING_INTERVAL, DEFAULT_RESUME_TRIES,
 	    DEFAULT_RESUME_DELAY);
 }
@@ -1527,6 +1529,9 @@ static int suricatta_settings(void *elem, void  __attribute__ ((__unused__)) *da
 	GET_FIELD_STRING_RESET(LIBCFG_PARSER, elem, "proxy", tmp);
 	if (strlen(tmp))
 		SETSTRING(channel_data_defaults.proxy, tmp);
+	GET_FIELD_STRING_RESET(LIBCFG_PARSER, elem, "artifactsstorage", tmp);
+	if (strlen(tmp))
+		SETSTRING(server_hawkbit.update_artifacts_storage_path, tmp);
 
 	return 0;
 
@@ -1586,7 +1591,7 @@ server_op_res_t server_start(char *fname, int argc, char *argv[])
 
 	/* reset to optind=1 to parse suricatta's argument vector */
 	optind = 1;
-	while ((choice = getopt_long(argc, argv, "t:i:c:u:p:xr:y::w:",
+	while ((choice = getopt_long(argc, argv, "t:i:c:u:p:xr:y::w:s:",
 				     long_options, NULL)) != -1) {
 		switch (choice) {
 		case 't':
@@ -1656,6 +1661,9 @@ server_op_res_t server_start(char *fname, int argc, char *argv[])
 			channel_data_defaults.retry_sleep =
 			    (unsigned int)strtoul(optarg, NULL, 10);
 			break;
+		case 's':
+			SETSTRING(server_hawkbit.update_artifacts_storage_path, optarg);
+			break;
 		case '?':
 		default:
 			return SERVER_EINIT;
diff --git a/suricatta/server_hawkbit.h b/suricatta/server_hawkbit.h
index f8ea5b8..3a2c6ea 100644
--- a/suricatta/server_hawkbit.h
+++ b/suricatta/server_hawkbit.h
@@ -38,6 +38,7 @@ typedef struct {
 	char *device_id;
 	char *tenant;
 	unsigned int polling_interval;
+	char *update_artifacts_storage_path;
 	bool polling_interval_from_server;
 	bool debug;
 	struct dictlist configdata;
-- 
2.9.4

